internal/backend/grpc/router/router.go

```diff
@@ -173,10 +173,28 @@ func (r *Router) Director(ctx context.Context, fullMethodName string) (proxy.Mod
 func (r *Router) getTalosBackend(ctx context.Context, md metadata.MD) ([]proxy.Backend, error) {
 	clusterName := getClusterName(md)
 
+	nodeID := getNodeID(md)
+
 	id := fmt.Sprintf("cluster-%s", clusterName)
 
 	if clusterName == "" {
-		id = fmt.Sprintf("machine-%s", getNodeID(md))
+		machineSetNode, err := safe.ReaderGetByID[*omni.MachineSetNode](actor.MarkContextAsInternalActor(ctx), r.cosiState, nodeID)
```

Comment:
should we add an integration test for all the ways we have right now?

Reply:
we already had one, it just might need to be updated

Reply:
yeah, will do that